src/python/WMCore/Services/Requests.py

```diff
@@ -410,8 +411,16 @@ def _getURLOpener(self):
 
     def addBasicAuth(self, username, password):
         """Add basic auth headers to request"""
-        auth_string = "Basic %s" % base64.encodestring('%s:%s' % (
+        ## TODO: base64.encodestring is deprecated
+        # https://docs.python.org/3.8/library/base64.html#base64.encodestring
+        # change to base64.encodebytes after we drop python2
+        username = encodeUnicodeToBytes(username)
+        password = encodeUnicodeToBytes(password)
+        encodedauth = base64.encodestring(b'%s:%s' % (
```

Comment:
Dario, here you lost me. A few lines above, you say this `encodestring` has been deprecated in python3. Shouldn't we then either remove it or only execute this code for PY2 environment?

Reply:
Well, they say that `base64.encodedstring` is a deprecated alias for `base64.encodedbytes`, but it is still available as an alias in python3.8. I would keep it like this at the moment, then when we are over the migration we can move to `base64.encodedbytes` (which is not available in py2). Would you rather have an `if PY3` here to select already now the correct alias based on the python runtime version?

Reply:
Your plan sounds good to me. If it works in python3.8, then we are good for the moment.

Reply:
Good, perfect, thank you!